src/parser/expression.js

```diff
@@ -753,6 +769,10 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
     node.properties.push(prop);
   }
 
+  if (firstRestLocation !== null) {
```

Comment:
no it has to be at the end after the loop, because otherwise the multiple destructuring check does not work.

Basically the logic is:
```
loop through all params:
   if spread
       if havent seen spread before AND this is not the end of the object pattern: record that we have seen a spread
       else if havent seen spread before: warning that this is already the second spread

if we finished the loop and recorded a spread: error that it was not at the end
```